src/sentry/preprod/size_analysis/download.py

```diff
@@ -0,0 +1,84 @@
+import logging
+
+from django.http.response import FileResponse
+from rest_framework.response import Response
+
+from sentry.models.files.file import File
+from sentry.preprod.models import PreprodArtifact, PreprodArtifactSizeMetrics
+
+logger = logging.getLogger(__name__)
+
+
+class SizeAnalysisError(Exception):
+    def __init__(self, message: str, status_code: int):
+        self.message = message
+        self.status_code = status_code
+        super().__init__(message)
+
+
+class SizeAnalysisMultipleResultsError(SizeAnalysisError):
+    def __init__(self, message: str = "Multiple size analysis results found for this artifact"):
+        super().__init__(message, 409)
+
+
+class SizeAnalysisInternalError(SizeAnalysisError):
+    def __init__(
+        self, message: str = "An internal error occurred while retrieving size analysis results"
+    ):
+        super().__init__(message, 500)
+
+
+class SizeAnalysisNotFoundError(SizeAnalysisError):
+    def __init__(self, message: str = "Size analysis not found"):
+        super().__init__(message, 404)
+
+
+def get_size_metrics_for_artifact(artifact: PreprodArtifact) -> PreprodArtifactSizeMetrics:
```

Comment:
Forgot about PreprodArtifact.get_size_metrics, will replace in commit coming shortly